src/diffusers/pipelines/pixart_alpha/pipeline_pixart_sigma.py

```diff
@@ -0,0 +1,430 @@
+# Copyright 2024 PixArt-Sigma Authors and The HuggingFace Team. All rights reserved.
```

Comment:
Help me understand the difference between the PixArt-Alpha and the PixArt-Sigma pipeline implementations? This way, we can decide if we really need to introduce a separate pipeline class here.

Reply:
The main difference is PixArt-Sigma doesn't have micro-conditions and PixArt-Sigma's transformer will use the qk_norm and key/value compression later. Therefore, the pipeline will accept these two later.

Reply:
is this the only difference? we can add a if else statement in pixart-alpha pipeline to support this 

```python
# 6.1 Prepare micro-conditions.
        added_cond_kwargs = {"resolution": None, "aspect_ratio": None}
        if self.transformer.config.sample_size == 128:
            resolution = torch.tensor([height, width]).repeat(batch_size * num_images_per_prompt, 1)
            aspect_ratio = torch.tensor([float(height / width)]).repeat(batch_size * num_images_per_prompt, 1)
            resolution = resolution.to(dtype=prompt_embeds.dtype, device=device)
            aspect_ratio = aspect_ratio.to(dtype=prompt_embeds.dtype, device=device)

            if do_classifier_free_guidance:
                resolution = torch.cat([resolution, resolution], dim=0)
                aspect_ratio = torch.cat([aspect_ratio, aspect_ratio], dim=0)

            added_cond_kwargs = {"resolution": resolution, "aspect_ratio": aspect_ratio}
```

Reply:
That's not the only difference. See: https://github.com/huggingface/diffusers/pull/7654/files#r1562377506

Reply:
PixArt-Sigma will have `if self.transformer.config.sample_size == 128:` as well but it doesn't use the addition conditions such as resolution and aspect ratio. But it could still be checked with `self.transformer.config.use_additional_condition` I believe?

Reply:
@lawrence-cj 

> PixArt-Sigma's transformer will use the qk_norm and key/value compression later. 

This will affect the Transformer block, right and the not pipeline itself. So, I guess that's fine to club it in a single pipeline?